float64 current_x
float64 current_y
---
float64 next_x
float64 next_y